Fluid/Filters/MiscFilters.cs

```diff
@@ -100,7 +101,32 @@ public static FilterCollection WithMiscFilters(this FilterCollection filters)
         /// </summary>
         public static ValueTask<FluidValue> Handleize(FluidValue input, FilterArguments arguments, TemplateContext context)
         {
-            return new StringValue(HtmlCaseRegex.Replace(input.ToStringValue(), "-$1$2").ToLowerInvariant());
+            var kebabCase = HtmlCaseRegex.Replace(input.ToStringValue(), "-$1$2").ToLowerInvariant();
+            var value = new StringBuilder(kebabCase);
+            var previousChar = value[0];
+
+            for (int i = 1; i < value.Length; i++)
+            {
+                var currentChar = value[i];
+                
+                if (Char.IsWhiteSpace(currentChar) || Char.IsPunctuation(currentChar))
+                {
+                    value.Replace(currentChar, KebabCaseSeparator);
```

Comment:
This is doing a replace on the whole string for every occurrence of an unsupported char?

Reply:
I didn't write that. @hishamco ?

Reply:
Seems I didn't check the whole code @deanbarker, I updated the PR

Reply:
I will reduce the number of removal by one for **n** consequence similar chars